File: reverseStringStack.cpp
#include<bits/stdc++.h>
using namespace std; 
void reverse(char*ch){
    stack<char>st;
    int n = strlen(ch);
    for(int i=0;i<n;++i)
        st.push(ch[i]); 
    for(int i=0;i<n;++i){
        ch[i] = st.top();
        st.pop();
    }
}
// void swap(char*a, char*b){
//     char temp =*a;
//     *a = *b;
//     *b = temp;
// }
// void reverse(char ch[]){
//     int n = strlen(ch),i;
//     for(i=0;i<n/2;++i)
//         swap(&ch[i],&ch[n-i-1]);
// }
int main(){
    char ch[] = "abcd";
    // reverse(ch);
    // cout << "Abrka Dabra: " << ch;
    reverse(ch);
    cout << ch << endl;
    return 0;
}

File: reverseStringStack.java
import java.util.*; 
class Stack{
    int size;
    int top;
    char[]a;
    boolean isEmpty(){
        return top < 0;
    }
    Stack(int n){
        top = -1;
        size = n;
        a = new char[size];        
    }
    void push(char x){
        if(top >= size){
            System.out.println("Stack overFlow");
            return;
        }
        a[++top] = x;
    }
    char pop(){
        if(top < 0){
            System.out.println("Stack underFlow");
            return 0;
        }
        return (char)a[top--];
    }
};
class reverseStringStack
{
    public static void reverse(StringBuffer str){
        int n = str.length();
        Stack stack = new Stack(n);

        for(int i=0;i<n;++i)
            stack.push(str.charAt(i));
        for(int i=0;i<n;++i)
        {  
            char ch = stack.pop(); 
            str.setCharAt(i,ch); 
        } 

    }
    public static void main(String args[]) 
    { 
        //create a new string 
        StringBuffer s= new StringBuffer("GeeksQuiz"); 
          
        //call reverse method 
        reverse(s); 
          
        //print the reversed string 
        System.out.println("Reversed string is " + s); 
    }       
}